[te] Disable flaky test CudaSharedMemReduce_1

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44862 [te] Disable flaky test CudaSharedMemReduce_1**

Differential Revision: [D23753831](https://our.internmc.facebook.com/intern/diff/D23753831)

bertmaher committed:
[te] Disable flaky test CudaSharedMemReduce_1

[ghstack-poisoned]

bertmaher commented:
This appears to be flaky failing on PRs/master, e.g.: https://app.circleci.com/pipelines/github/pytorch/pytorch/214990/workflows/f617cd89-c848-437f-8d44-d0d640b22277/jobs/7530976/steps.  (I guessed that it's this specific test due to the value of b_ref in the failure).  It suggests a problem with reduction codegen which is worrisome but it's not immediately obvious to me.  So, I'm going to disable this one so it won't fail people's PRs, and we can work on solving it.

ZolotukhinM commented on test/cpp/tensorexpr/tests.h:
```diff
@@ -443,6 +442,9 @@ namespace jit {
   _(CudaMaskMultiDimMultiAxis)             \
   _(CudaMaskMultiDimMultiLevel)
+  _(CudaMaskThreadDim)
+// _(CudaMaskBlockAndThreadDim)
```
What's up with `CudaMaskBlockAndThreadDim`?

ZolotukhinM reviewed (approved):
Approving to make our CI great again!

bertmaher committed:
Update on "[te] Disable flaky test CudaSharedMemReduce_1"


Differential Revision: [D23753831](https://our.internmc.facebook.com/intern/diff/D23753831)

[ghstack-poisoned]
